tools/setup.rb

```diff
@@ -0,0 +1,536 @@
+#!/usr/bin/env ruby
```

Comment:
Same here, I'd prefer this be moved into a `docker` directory so it's clear what it's for.  `setup.rb` is kind of vague.